Module 1: 1. sum of n numbers

In [1]:
# Step 1: Input the value of n
n = int(input("Enter the value of n: "))

# Step 2: Initialize an empty list 'numbers'
numbers = []

# Step 3: Input n numbers and add them to the list
for i in range(n):
    number = float(input(f"Enter number {i+1}: "))
    numbers.append(number)

# Step 4: Initialize 'sum' as 0
sum = 0

# Step 5: Calculate the sum
for num in numbers:
    sum += num

# Step 6: Print the sum
print("Sum of the numbers:", sum)


Sum of the numbers: 150.0


OUTPUT:
=======

Enter the value of n: 5
Enter number 1: 10
Enter number 2: 20
Enter number 3: 30
Enter number 4: 40
Enter number 5: 50
Sum of the numbers: 150


Module 1: 2 Factorial of N 

In [1]:
# Step 1: Input the value of 'n'
n = int(input("Enter a positive integer (n) to calculate its factorial: "))

# Step 2: Initialize 'factorial' as 1
factorial = 1

# Step 3: Calculate the factorial
for i in range(1, n + 1):
    factorial *= i

# Step 4: Print the factorial
print(f"The factorial of {n} is: {factorial}")


The factorial of 5 is: 120


Module 1: 3 Matrix multiplication

OUTPUT
======

Enter a positive integer (n) to calculate its factorial: 5
The factorial of 5 is: 120


Enter elements of matrix A:
Enter elements of matrix B:
Result of matrix multiplication (C):
58 64 
139 154 


OUTPUT: 
=======
Enter the number of rows for the first matrix: 2
Enter the number of columns for the first matrix: 3
Enter the number of rows for the second matrix: 3
Enter the number of columns for the second matrix: 2
Enter elements of matrix A:
1 2 3
4 5 6
Enter elements of matrix B:
7 8
9 10
11 12
Result of matrix multiplication (C):
58 64 
139 154


Module 1: 4 Insertion sort

In [7]:
# Step 1: Input an array of numbers
arr = list(map(int, input("Enter an array of numbers separated by spaces: ").split()))

# Step 2: Implement Insertion Sort
for i in range(1, len(arr)):
    key = arr[i]
    j = i - 1

    while j >= 0 and key < arr[j]:
        arr[j + 1] = arr[j]
        j -= 1

    arr[j + 1] = key

# Step 3: Print the sorted array
print("Sorted Array:", arr)


Sorted Array: [5, 6, 11, 12, 13]


OUTPUT: 
=======

Enter an array of numbers separated by spaces: 12 11 13 5 6
Sorted Array: [5, 6, 11, 12, 13]
